contracts/lib/CampaignLibrary.sol

```diff
@@ -0,0 +1,23 @@
+pragma solidity ^0.4.19;
+
+
+library CampaignLibrary {
+
+    struct Filters {
+        string countries;
+        string packageName;
+        uint[] vercodes;
+    }
+
+    struct Campaign {
+        bytes32 bidId;
+        uint price;
+        uint budget;
+        uint startDate;
+        uint endDate;
+        bool valid;
+        address  owner;
+        Filters filters;
+        string ipValidator;
+    }
```

Comment:
I think we should implement getters and setters here for Campaigns and Filters and use this getters and setters in the AdvertisementStorage instead of accessing directly to structure elements, what do you think?

Reply:
@tcrosado I already answered that one on the top.

> We cannot use the library for all the Getters and Setters of the struct. It makes it impossible to deploy (runs out of gas due to the call to the library -> +5mm gas used). The fact that a campaign is public, we can access its attributes all the time.

Reply:
@GustavoRSSilva I've read everything and I agree with most of it. The only thing I'm failing to understand is how can we access a campaign without a getters. It's a struct and since a struct can't be returned, even if it would be public in the contract (which isn't the case at the moment), there's no way to access it right?

Reply:
@DioPires the struct is just how we arrange data. all of his attributes are public.

![screenshot from 2018-05-29 11-10-32](https://user-images.githubusercontent.com/8384988/40652780-228aef12-6331-11e8-9dc7-265aac80f505.png)

![screenshot from 2018-05-29 11-16-25](https://user-images.githubusercontent.com/8384988/40652992-c22476e2-6331-11e8-85da-13f00cf139a7.png)


There are two files that were developed, the Campaign library and the Campaign storage.
The library is where the campaign struct defined, sets the startData as uint, the countries a string, etc.
The storage is where the campaign in saved.

We use an library because we want to use the same struct in the campaign and in the campaign Storage. Keep the data consistent.

using Getters and Setters when all the attributes are public was adding to much weight to the project makingit impossible to deploy.


Keep in mind that we cannot send struct between contracts ( ** language limitation ** ), so we need to send the data and encode or decode on the other side.